### PR TITLE
Recipe to Replace `sun.misc.BASE64Encoder/Decoder` with `java.util.Base64.Encoder/Decoder`

### PR DESCRIPTION
**Pull request by BhavanaPidapa**
## What's changed?
This recipe deals with a part of this rule
image

## What's your motivation?
I used org.openrewrite.java.ChangeType for the recipe RemovedSunAPIs

## Anyone you would like to review specifically?
@cjobinabo @timtebeek 

## Have you considered any alternatives or workarounds?
Since it was not possible to test the recipe. I created sample files, created a local plugin using
./gradlew publishToMavenLocal then ran it in the Java8 Sample App using mvn rewrite:dryRun

Attaching the rewrite.patch file
[rewrite.patch](https://github.com/user-attachments/files/15905680/rewrite.patch)

## Any additional context
This recipe covers only `The sun.* APIs are unsupported, JDK-internal APIs, and may go away at any time. The sun.misc.BASE64Encoder and sun.misc.BASE64Decoder have been removed in Java SE 11. Instead, use the supported java.util.Base64 class, which was added in Java SE 8.` part of the rule.

### Checklist
- [X] I've added unit tests to cover both positive and negative cases
- [X] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [X] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Comment by timtebeek** [comment]
Doesn't this overlap with this existing recipe?
https://github.com/openrewrite/rewrite-migrate-java/blob/86bc2b56109aa64eb68d23c31d7565821f7ab468/src/main/java/org/openrewrite/java/migrate/UseJavaUtilBase64.java#L63-L72

File: src/test/java/org/openrewrite/java/migrate/Java8ToJava11Test.java [hunk 1]
```diff
@@ -105,4 +122,40 @@ void bar() {
           )
         );
     }
+
+    @Test
+    void Krb5LoginModuleTest() {
```
Comment (cjobinabo) [inline src/test/java/org/openrewrite/java/migrate/Java8ToJava11Test.java:127]: These tests don't appear to be related to your pull request. Perhaps they were merged by mistake

**Comment by cjobinabo** [comment]
> Doesn't this overlap with this existing recipe?
> 
> https://github.com/openrewrite/rewrite-migrate-java/blob/86bc2b56109aa64eb68d23c31d7565821f7ab468/src/main/java/org/openrewrite/java/migrate/UseJavaUtilBase64.java#L63-L72

Great find. The recipes do appear to overlap. We can opt for the suggestion mentioned in [UseJavaUtilBase64.java](https://github.com/openrewrite/rewrite-migrate-java/blob/86bc2b56109aa64eb68d23c31d7565821f7ab468/src/main/java/org/openrewrite/java/migrate/UseJavaUtilBase64.java#L63-L72) instead

**Comment by cjobinabo** [comment]
@BhavanaPidapa feel free to close this PR when you are ready

**Comment by BhavanaPidapa** [comment]
With the existing custom recipe, I have tested all the test cases and results are as expected. So, I am closing this PR.
